[CUBRIDMAN-80] Reflect the changed TRUNCATE specification (#297)
CUBRID/cubrid-manual: en/sql/query/truncate.rst

You can delete all records in the specified table by using the **TRUNCATE** statement.

> It has following advantages than using the **DELETE FROM** *table_name* statement without a **WHERE** clause.

Reviewer: ```suggestion
It has the following advantages over using the **DELETE FROM** *table_name* statement without a **WHERE** clause.
```

> * It's way faster due to deleting all indexes and constraints in adavance and deleing records at once.

Reviewer: ```suggestion
* It's way faster due to deleting all indexes and constraints in advance and deleting records at once.
```
typo corrections of 'advance' and 'deleting'.

* There is no vacuum cost.

> * It generates way lesser log records so that it is better in terms of HA, Recovery and Rollback.

Reviewer: ```suggestion
* It generates way lesser log records so that it is better in terms of HA, Recovery, and Rollback.
```

> * It generates way lesser log records so that it is better in terms of HA, Recovery and Rollback.

Reviewer 2: ```suggestion
* It generates way lesser log records so that it is better in terms of HA replication, Recovery, and Rollback.
```

.. note:: 

>     * The **ON DELETE** trigger is disabled while deleting records with the **TRUNCATE** statement.

Reviewer 2: Does this sentence mean that the trigger does not work when the truncate statement is executed? 
Or does it mean that the ON DELETE action does not work if only the truncate statement is executed without CASCADE ?

Reviewer 2: If what you mean is a deleting trigger, you will modify it as the following.
* The deleting trigger is disabled while deleting records with the **TRUNCATE** statement.

Otherwise, it would be better to modify the trigger to an action as the following.
* The **ON DELETE** action is disabled while deleting records with the **TRUNCATE** statement.

Author: It means trigger. I'll follow your instruction. Thank you.

Author: How about **DELETE** trigger? which seems to be what the manual call it.

Author: @mhoh3963 How about **DELETE** trigger? which seems to be what the manual calls it.

Reviewer 2: OK, it is good.

    * The **TRUNCATE** statement initializes the **AUTO INCREMENT** column of the table. Therefore, if data is inserted, the **AUTO INCREMENT** column value increases from the initial value. 
    * To execute the **TRUNCATE** statement, the authorization of **ALTER**, **INDEX**, and **DELETE** is required on the table. For granting authorization, see :ref:`granting-authorization`.
    * If a table is created with **DONT_REUSE_OID** option (:ref:`dont-reuse-oid`), other tables can refer to it as a column. When the specified table is referred to like this, it is impossible to truncate the table and it's done with **DELETE FROM** statement internally.

::

    TRUNCATE [ TABLE ] <table_name>

*   *table_name* : Specifies the name of the table that contains the data to be deleted.

> *   **CASACDE** : Deletes all records in all tables referring to the specified table with a foreign key. This is propagated to all tables in foreign key relationship. A **PRIMARY KEY** constraint has to be defined in the table and this has to be referred to by one or more **FOREIGN KEY**, and the foreign key action has to be **ON DELETE**. It fails without this option when a foreign key referring to the specified table is defined. It also fails when even one **ON DELETE** action is not **CASCADE** in all foreign key relationships. See :ref:`foreign-key-constraint` for more information about foreign key constraint. 

Reviewer: ```suggestion
*   **CASACDE** : Deletes all records in all tables referring to the specified table with a foreign key. This is propagated to all tables in the foreign key relationship. A **PRIMARY KEY** constraint has to be defined in the table and this has to be referred to by one or more **FOREIGN KEY**, and the foreign key action has to be **ON DELETE**. It fails without this option when a foreign key referring to the specified table is defined. It also fails when even one **ON DELETE** action is not **CASCADE** in all foreign key relationships. See :ref:`foreign-key-constraint` for more information about the foreign key constraint. 
```

> *   **CASACDE** : Deletes all records in all tables referring to the specified table with a foreign key. This is propagated to all tables in foreign key relationship. A **PRIMARY KEY** constraint has to be defined in the table and this has to be referred to by one or more **FOREIGN KEY**, and the foreign key action has to be **ON DELETE**. It fails without this option when a foreign key referring to the specified table is defined. It also fails when even one **ON DELETE** action is not **CASCADE** in all foreign key relationships. See :ref:`foreign-key-constraint` for more information about foreign key constraint. 

Reviewer 2: ```suggestion
*   **CASCADE** : Deletes all records in all tables referring to the specified table with a foreign key. This is propagated to all tables in foreign key relationship. A **PRIMARY KEY** constraint has to be defined in the table and this has to be referred to by one or more **FOREIGN KEY**, and the foreign key action has to be **ON DELETE**. It fails without this option when a foreign key referring to the specified table is defined. It also fails when even one **ON DELETE** action is not **CASCADE** in all foreign key relationships. See :ref:`foreign-key-constraint` for more information about foreign key constraint. 
```

Author: oh my god... thank you

                a
    =============
                3

> The following example uses the **CASCADE** option. Note that a record even refers to nothing, but the propagated **TRUNCATE** deletes all records.                 

Reviewer: ```suggestion
The following example uses the **CASCADE** option. Note that even a record refers to nothing, but the propagated **TRUNCATE** deletes all records.                 
```
